api/src/transactions.rs

```diff
@@ -639,10 +639,23 @@ impl TransactionsApi {
                     vec![signed_transaction.sender().to_vec()],
                     context.node_config.api.max_gas_view_function,
                 );
-                let values = output.values.map_err(|err| {
-                    SubmitTransactionError::bad_request_with_code_no_info(
-                        err,
+                let values = output.values.map_err(|status| {
```

Comment:
nit: seems duplicated with `api/src/view_function.rs`, can be refactored into a single method?